calendar/store/welcome_store_test.go

```diff
@@ -0,0 +1,189 @@
+package store
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/tracker/mock_tracker"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/utils/bot/mock_bot"
+)
+
+type MockPluginAPI struct {
+	plugin.API
+	mock.Mock
+}
+
+func (m *MockPluginAPI) KVGet(key string) ([]byte, *model.AppError) {
+	args := m.Called(key)
+	data, _ := args.Get(0).([]byte)
+	if err := args.Get(1); err != nil {
+		return nil, err.(*model.AppError)
+	}
+	return data, nil
+}
```

Comment:
We have created this in another testcases file as well. Can we make it generalised

Reply:
can do but we will need to change the base of this PR to `MM-843`